app/actions/nodeStorageActions.js

```diff
@@ -15,19 +15,44 @@ import { findNetworkByLabel } from '../core/networks'
 
 const ID = 'nodeStorage'
 const STORAGE_KEY = 'selectedNode'
+const CACHE_EXPIRATION =
+  15 /* minutes */ * 60 /* seconds */ * 1000 /* milliseconds */
 const cachedRPCUrl = {}
 
+type Net = NetworkLabelTypes
+
 type Props = {
   url: string,
-  net: string
+  net: Net
+}
+
+export const determineIfCacheIsExpired = (
+  timestamp: number,
+  expiration: number = CACHE_EXPIRATION
+): boolean => timestamp + expiration < new Date().getTime()
+
+export const parseUrlData = (data: {
+  height: number,
+  port: string,
+  protocol: string,
+  url: string
+}): string => {
+  let url = data.protocol ? `${data.protocol}://${data.url}` : data.url
+  url = data.port ? `${url}:${data.port}` : url
+  return url
```

Comment:
I think the function name is a bit confusing, it's more like `buildURL`

Regarding the implementation, how about using lodash compact and an array that you join? (https://lodash.com/docs/4.17.11#compact)


```
const { protocol, url, port } = url

return compact([
  protocol && `${protocol}://`,
  url,
  port && `:${port}`
]).join('')
```